### ISSUE TITLE
Logging Documentation Does not Detail How to Access the Logged Values during the fit loop 

### ISSUE THREAD

**Issue by hyang0129** [Lightning-AI/pytorch-lightning#19758]
### 📚 Documentation

So https://lightning.ai/docs/pytorch/stable/extensions/logging.html explains how to add loggers and log the information. I cannot find in the docs how to access said information within the training loop. For example, I want to print the epoch loss at the end of the epoch. If the loss is already logged, it would make sense to access the logged info and pick it up from there. 

I assume its possible because the logger must store the info somewhere. I tried trainer.logged_metrics but that attribute isn't present.

**Comment by SkafteNicki** [comment]
Hi, thanks for raising this issue. Not sure why you are observing that `trainer.logged_metrics` is not available, because that is exactly the property to get the latest metrics logged in `self.log`/`self.log_dict`. A simple example where the values are printed in a callback:

```python
import torch
import torch.nn as nn
import torch.nn.functional as F
from torch.utils.data import DataLoader, TensorDataset
import pytorch_lightning as pl
from pytorch_lightning.callbacks import Callback


class SimpleModel(pl.LightningModule):
    def __init__(self):
        super().__init__()
        self.layer = nn.Linear(10, 1)
        
    def forward(self, x):
        return self.layer(x)
    
    def training_step(self, batch, batch_idx):
        x, y = batch
        loss = F.mse_loss(self(x), y)
        self.log("train_loss", loss)
        return loss
    
    def configure_optimizers(self):
        return torch.optim.Adam(self.parameters(), lr=0.001)


class PrintMetricsCallback(Callback):
    def on_train_epoch_end(self, trainer, pl_module):
        print(f"End of epoch {trainer.current_epoch}:")
        print(f"  trainer.logged_metrics: {trainer.logged_metrics}")
        print(f"  trainer.callback_metrics: {trainer.callback_metrics}")


# Create dummy data and model
x, y = torch.randn(100, 10), torch.randn(100, 1)
dataloader = DataLoader(TensorDataset(x, y), batch_size=20)
model = SimpleModel()

# Train with callback that accesses logged metrics
trainer = pl.Trainer(max_epochs=2, callbacks=[PrintMetricsCallback()], enable_progress_bar=False)
trainer.fit(model, dataloader)

print(f"\nFinal verification:")
print(f"trainer.logged_metrics exists: {hasattr(trainer, 'logged_metrics')}")
print(f"trainer.logged_metrics: {trainer.logged_metrics}")
```

The documented properties of trainer also mentions this:
https://lightning.ai/docs/pytorch/stable/common/trainer.html#logged-metrics

Closing issue, but feel free to ping and reopen if necessary.
